src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -951,14 +951,55 @@ private void buildDockerImage(File tempDockerfile, File userDockerfile) throws P
             debug("Docker build context: " + userDockerfile.getParent());
             buildCmd = "docker build -f " + tempDockerfile + " -t " + imageName + " " + userDockerfile.getParent();
             info(buildCmd);
-            String buildOutput = execDockerCmd(buildCmd, dockerBuildTimeout);
-            debug("Docker build output: " + buildOutput);
+            long startTime = System.currentTimeMillis();
+            runCommandAndLog(getRunProcess(buildCmd), dockerBuildTimeout);
+            checkDockerIgnore(startTime, userDockerfile);
+            info("Completed building Docker image.");
+        } catch (IllegalThreadStateException  e) {
+            // the timeout was too short and the docker command has not yet completed.
+            debug("IllegalThreadStateException, message="+e.getMessage());
+            error("The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds. " +
+                "Use the dockerBuildTimeout option to specify a longer period or " +
+                "add files not needed in the container to the .dockerignore file.", e);
+            throw new PluginExecutionException("Could not build Docker image using Dockerfile: " + 
+                userDockerfile.getAbsolutePath() + ". Address the following docker build error and then start dev mode again: " +
+                "The docker build command did not complete within the timeout period: " + dockerBuildTimeout + " seconds", e);
+        } catch (IOException e) {
+            error("Input or output error building Docker image: " + e.getMessage());
+            throw new RuntimeException(e);
+        } catch (InterruptedException e) {
+            error("Thread was interrupted while building the Docker image: " + e.getMessage());
+            throw new PluginExecutionException("Could not build Docker image using Dockerfile: " + userDockerfile.getAbsolutePath() + ". Address the following docker build error and then start dev mode again: " + e.getMessage(), e);
```

Comment:
Is it the same as above here? Would `e` be the docker build error or some other Exception?

Reply:
I just wanted to capture the interrupted exception in case that was relevant. This is better done with a debug message and then the standard exit.